src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -115,6 +126,54 @@ public function onKernelResponse(FilterResponseEvent $event)
              * it is saved will just restart it.
              */
             $session->save();
+
+            /*
+             * For supporting sessions in php runtime with runners like roadrunner or swoole the session
```

Comment:
Shouldn't it be done by swoole (or roadrunner)'s middleware then?

Reply:
The problem is that swoole and roadrunner doesn't no anything about the symfony framework session configuration and so can not handle this correctly based on that configuration. To avoid that we need to duplicate also the logic in the php-runtime swoole and roadrunner implementation I thought it would be great if Symfony Sessions just work out of the box running in CLI context.

Reply:
Also we are not knowing in the middlewares if a session was started and saved which is important for correct handling in all cases.

Reply:
Did rebase the PR and target the 5.4 branch now.